gsl/span

```diff
@@ -359,6 +359,7 @@ public:
     constexpr static const index_type extent = Extent;
 
     // [span.cons], span constructors, copy, assignment, and destructor
+    template <bool Unused = false, class = std::enable_if_t<(Unused || Extent <= 0)>>
```

Comment:
Since `Extent <= 0` does not depend on a parameter of this function template, `std::enable_if_t<Extent <= 0>` is ill-formed when `Extent` is greater than `0`. If we tried to constrain this constructor template on simply `std::enable_if_t<Extent <= 0>` the compiler would rightfully complain about `std::enable_if<false>` having no member named `type`. `Unused || Extent <= 0`, on the other hand, *is* an expression that depends on a template parameter. If, when the compiler tries to substitute `false` for `Unused` in `std::enable_if_t<(Unused || Extent <= 0)>` and discovers that `std::enable_if<false>` has no member named `type`, SFINAE kicks in and removes the overload from the set of viable candidates rather than reporting an error.

(`Unused` is actually a poor name here. `Dependent` is a little more suggestive. I'll change it, and leave a more concise comment in the code.)